torch/nn/_functions/packing.py

```diff
@@ -14,15 +14,14 @@ def forward(ctx, input, lengths, batch_first):
 
         steps = []
         batch_sizes = []
-        lengths_iter = reversed(lengths)
         batch_size = input.size(1)
 
         if len(lengths) != batch_size:
             raise ValueError("Expected `len(lengths)` to be equal to batch_size, but got "
                              "{} (batch_size={}).".format(len(lengths), batch_size))
 
         prev_l = 0
-        for i, l in enumerate(lengths_iter):
+        for i, l in enumerate(reversed(list(lengths))):
```

Comment:
Was there are a particular reason `lengths_iter` got inlined?

Reply:
an attempted clean-up that I ended up changing almost-back. I'll change it back all the way